lightning/src/ln/chan_utils.rs

```diff
@@ -217,87 +217,100 @@ pub fn derive_public_key<T: secp256k1::Signing>(secp_ctx: &Secp256k1<T>, per_com
 
 /// Derives a per-commitment-transaction revocation key from its constituent parts.
 ///
+/// Only the transaction broadcaster owns a valid witness to propagate a revoked
+/// commitment transaction, thus per_commitment_secret always come from broadcaster
+/// and revocation_base_secret always come from countersignatory of the transaction.
+///
 /// Note that this is infallible iff we trust that at least one of the two input keys are randomly
 /// generated (ie our own).
-pub fn derive_private_revocation_key<T: secp256k1::Signing>(secp_ctx: &Secp256k1<T>, per_commitment_secret: &SecretKey, revocation_base_secret: &SecretKey) -> Result<SecretKey, secp256k1::Error> {
-	let revocation_base_point = PublicKey::from_secret_key(&secp_ctx, &revocation_base_secret);
-	let per_commitment_point = PublicKey::from_secret_key(&secp_ctx, &per_commitment_secret);
+pub fn derive_private_broadcaster_revocation_key<T: secp256k1::Signing>(secp_ctx: &Secp256k1<T>, broadcaster_per_commitment_secret: &SecretKey, countersignatory_revocation_base_secret: &SecretKey) -> Result<SecretKey, secp256k1::Error> {
```

Comment:
Same here as below, I think we shouldn't refer to the revocation keys as "broadcaster" or "countersignatory" - they're not really "owned" by either side.